libraries/ESP8266WebServer/src/ESP8266WebServerSecure.cpp

```diff
@@ -0,0 +1,149 @@
+/*
+  ESP8266WebServerSecure.cpp - Dead simple HTTPS web-server.
+  Supports only one simultaneous client, knows how to handle GET and POST.
+
+  Copyright (c) 2017 Earle F. Philhower, III. All rights reserved.
+
+  This library is free software; you can redistribute it and/or
+  modify it under the terms of the GNU Lesser General Public
+  License as published by the Free Software Foundation; either
+  version 2.1 of the License, or (at your option) any later version.
+
+  This library is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+  Lesser General Public License for more details.
+
+  You should have received a copy of the GNU Lesser General Public
+  License along with this library; if not, write to the Free Software
+  Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+  Modified 8 May 2015 by Hristo Gochkov (proper post and file upload handling)
+*/
+
+
+#include <Arduino.h>
+#include <libb64/cencode.h>
+#include "WiFiServer.h"
+#include "WiFiClient.h"
+#include "ESP8266WebServerSecure.h"
+
+
+ESP8266WebServerSecure::ESP8266WebServerSecure(IPAddress addr, int port) 
+: _serverSecure(addr, port)
+{
+}
+
+ESP8266WebServerSecure::ESP8266WebServerSecure(int port)
+: _serverSecure(port)
+{
+}
+
+void ESP8266WebServerSecure::setServerKeyAndCert_P(const uint8_t *key, int keyLen, const uint8_t *cert, int certLen)
+{
+    _serverSecure.setServerKeyAndCert_P(key, keyLen, cert, certLen);
+}
+
+void ESP8266WebServerSecure::setServerKeyAndCert(const uint8_t *key, int keyLen, const uint8_t *cert, int certLen)
+{
+    _serverSecure.setServerKeyAndCert(key, keyLen, cert, certLen);
+}
+
+ESP8266WebServerSecure::~ESP8266WebServerSecure() {
+  if (_currentHeaders)
+    delete[]_currentHeaders;
+  _headerKeysCount = 0;
+  RequestHandler* handler = _firstHandler;
+  while (handler) {
+    RequestHandler* next = handler->next();
+    delete handler;
+    handler = next;
+  }
+  close();
```

Comment:
if the destructor is made virtual, ESP8266WebServerSecure will not need to do parts of the cleanup which are done by ESP8266WebServer.

Reply:
Done.  Did not know that, thought the vtable only had the final destructor.  Learn something new every day!